concrete-core/src/specification/engines/cleartext_creation.rs

```diff
@@ -0,0 +1,33 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::CleartextEntity;
+
+engine_error! {
+    CleartextCreationError for CleartextCreationEngine @
+}
+
+/// A trait for engines creating cleartexts.
+///
+/// # Semantics
+///
+/// This [pure](super#operation-semantics) operation generates a cleartext from the `raw` value.
+///
+/// # Formal Definition
+pub trait CleartextCreationEngine<Raw, Cleartext>: AbstractEngine
+where
+    Cleartext: CleartextEntity,
+{
+    /// Creates a cleartext from a raw value.
+    fn create_cleartext(
+        &mut self,
+        raw: &Raw,
+    ) -> Result<Cleartext, CleartextCreationError<Self::EngineError>>;
```

Comment:
Very good question !

Here a `Raw` value is expected to be mainly a `u32` or `u64`, but actually, depending on the needs of the backend implementors it could be something different:

Most of the operations described in this specification act on types defined inside of `concrete-core`. For instance, only in `concrete-core` can a type implement the `LweCiphertextEntity` trait for instance. 

Concrete (and FHE) is not a closed box though, and it must be possible to at least create a cleartext from any type that suits a user, and retrieve this type from the cleartext. Usually the types that suits users are u32 and u64, but that may be different in some cases (still a type representing an integer but maybe a different one ? I don't know).

For this reason I chose the `Raw` name for those types (which are any type that suit the backend implementor), and I chose not to put any bound on it. It could basically be any type.

Reply:
Discussing with @ilaria-zama and @jibe, we chose to replace `Raw` by `Value` here. As well as adding a small explanation about what this `Value` type is.